wordpress-mobile/WordPress-iOS
# Reader Comments: Enable the new comment threads.

Refs #17475
Depends on #17669

This PR enables the new comment threads to the public. Here's the design:

![145415445-3b843e0c-293a-4585-8526-1335f58487c6](https://user-images.githubusercontent.com/1299411/145715304-aec28574-e763-496a-98b7-17e823e3bad2.png)

⚠️ Note that in this iteration, the comment body will still be rendered using the previous mechanism, due to technical reasons. However, the rest of the components should be updated.

Some quick overview on the new features/capabilities:

- **Author label**. This is shown when the comment is written by the post author.
- **Accessory button** (the gray icon button on the right). Depending on your moderation capabilities, you could see an ellipsis icon or a share icon.
  - Tapping on the share icon should share the comment by presenting a share sheet.
- **Row highlighting**. When navigating to the comment thread from a comment notification, the row is highlighted to show that it's the one that you're reading in the Notifications screen.
- **Reply highlighting**. Tapping on the Reply button should now turn the button blue, and highlights the entire row. Tapping on the Reply button again should un-highlight the row.

## To Test

- Verify that the new comment thread is displayed correctly from these entry points:
  - Notifications > comment reply notification > tap on the notification header row.
  - Reader > tap the comments icon on the Reader post card.
  - Reader > select any post > tap on the comments icon.
  - My Site > Comments > select any reply comment > tap on the header row.
- Verify that the previous comment threads functionality work fine:
  - Submit a comment for the post.
  - Submit a reply to another comment.
  - Like or unlike the comment.
- Verify that the new features/functionalities work as described in the description section above 👆 . 

## Related PRs

- #17656
- #17669
- #17658
- #17635
- #17634
- #17629
- #17623
- #17622
- #17601
- #17600
- #17589
- #17566
- #17557
- #17551
- #17549
- #17535
- #17532
- #17517
- #17487
- #17480

## Regression Notes
1. Potential unintended areas of impact
Some features from the previous comment thread could be impacted.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested the features.

3. What automated tests I added (or what prevented me from doing so)
n/a.

PR submission checklist:
